### DISCUSSION TITLE
Add Prefix parameter to zoautil_py.jobs.listing

### DISCUSSION THREAD

**Discussion by elaborateballoons** [IBM/zoau-samples#63] [Ideas] [answered]
When using regular "jls", it is possible to find jobs on the system with the job prefix. I am not seeing this option in the corresponding python module zoautil_py.jobs.listing. The only parameter options available are job_id or job_owner. I tried both "prefix" and "job_prefix" in case it was a documentation error but both failed. It would be very useful to be able to list all jobs by prefix through this python module.

**Comment by EnsonoStevenPerva** [comment, marked as answer]
Here's a bit of sample code that should get you on your way. It's not as intuitive as you're describing it to be, and perhaps the takeaway is that ZOAU needs to keyword this for usability sake, but this is a bandage:

For example, if you were to run a listing looking for all jobs starting with an O you would typically run this:
`jls '/*/O*'`

giving me this result:
`OAMSTC   OAM      STC03939 AC       ?`
In Python, that would look like this:

```
from zoautil_py import jobs
val = jobs.listing('/*/O*')
print(val)
for job in val:
  print(job)
```

This code snippet gives me:
```
[STC03939]
[Job]
        id: STC03939
        name: OAM
        owner: OAMSTC
        status: AC
        rc: ?
```

Hope this helps!

**Reply by elaborateballoons** [reply]
Okay, great! Looking at the Python API documentation made it seem like those were the only two options but looking at the actual module code I can see how your example works. I would definitely have been on my way if this was spelled out somewhere in docs so I think adding it would be great. Thank you!

**Reply by EnsonoStevenPerva** [reply]
No trouble at all. You're now the expert! I have found that using some of the `_` methods (`jobs._listing(...)`) gives me a ZOAUResponse object which gives me some insight into how things get done under the hood. Once I learned how it was issuing the `jls` command, it was pretty simple to pivot to a working example for your specific use-case.

I do agree, I would prefer to have every possible option serialized as a canned kwarg, though. It would be much more usable, and allow the syntax under the hood to change without our code needing any manipulation over time. @AnthonyGiorgio any thoughts?
